r/R/array.R

```diff
@@ -328,3 +328,17 @@ is.Array <- function(x, type = NULL) { # nolint
   }
   is_it
 }
+
+#' @rdname array
+#' @usage NULL
+#' @format NULL
+#' @export
+MapArray <- R6Class("MapArray",
+  inherit = ListArray,
+  public = list(
+    keys = function() MapArray__keys(self),
+    items = function() MapArray__items(self),
+    keys_nested = function() MapArray__keys_nested(self),
+    items_nested = function() MapArray__items_nested(self)
```

Comment:
These methods don't exist in other implementations, but I thought they would be useful to R users.